addons/website/static/src/components/survey/survey.js

```diff
@@ -0,0 +1,514 @@
+        return Object.values(state.palettes);
+    },
+};
+
+async function getInitialState(wid) {
+
+    const features = rpc.query({
+        model: 'website.survey.feature',
+        method: 'search_read',
+        fields: ['title', 'description', 'type', 'icon', 'website_types_preselection'],
+    }).then(function (results) {
+        const features = {};
+        results.forEach(result => {
+            features[result.id] = Object.assign({}, result, {selected: false});
+            const wtp = features[result.id].website_types_preselection;
+            features[result.id].website_types_preselection = wtp ? wtp.split(',') : [];
+        });
+        return features;
+    });
+
+    const logo = rpc.query({
+        model: 'website',
+        method: 'get_survey_logo',
+        args: [[parseInt(wid)]],
+    });
+    const industries = rpc.query({
+        model: 'website',
+        method: 'get_survey_industries',
+    }).catch((_) => false);
+
+    const allPalettes = {}, palettes = {};
+    const style = window.getComputedStyle(document.documentElement);
+    const themeNames = weUtils.getCSSVariableValue('theme-names', style).split(' ');
+    themeNames.forEach((rawName) => {
+        const name = rawName.replace(/'/g, '');
+        const count = weUtils.getCSSVariableValue(`o-count-${name}`, style);
+        for (let i = 1; i <= count; i++) {
+            const paletteName = `${name}-${i}`;
+            const palette = {};
+            palette['id'] = paletteName;
+            for (let j = 1; j <= 5; j += 1) {
+                const color = weUtils.getCSSVariableValue(`o-palette-${name}-${i}-o-color-${j}`, style);
+                palette[`color${j}`] = color;
+            }
+            let duplicate = false;
+            for (const validatedPalette of Object.values(allPalettes)) {
+                if (validatedPalette.color1.toLowerCase() === palette.color1.toLowerCase() && validatedPalette.color2.toLowerCase() === palette.color2.toLowerCase()) {
+                    duplicate = true;
+                }
+            }
+            if (!duplicate) {
+                allPalettes[paletteName] = palette;
+                if (Object.keys(palettes).length < MAX_PALETTES) {
+                    palettes[paletteName] = allPalettes[paletteName];
+                }
+            }
+        }
+    });
+
+    const vals = await Promise.all([features, logo, industries]);
+    if (!vals[2]) {
+        await skipSurvey(parseInt(wid));
+        return false;
+    }
+
+    return {
+        selectedType: undefined,
+        selectedPurpose: undefined,
+        selectedIndustry: undefined,
+        industries: vals[2],
+        logo: vals[1],
+        selectedPalette: undefined,
+        recommendedPalette: undefined,
+        palettes: palettes,
+        allPalettes: allPalettes,
+        features: vals[0],
+        themes: Array(3).fill({
+            name: undefined,
+            id: undefined,
+            svg: undefined,
+        }),
```

Comment:
This will fill the array with 3 references to the same anonymous object, meaning they will share state, this is probably not what you want here.